pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -34,6 +34,12 @@ import (
 )
 
 func makeTenantSpan(tenantID uint64) roachpb.Span {
+	// TODO(dt): remove conditional if we make MakeTenantSpan do this.
```

Comment:
could you add a docstring which explains why this is necessary? 

Also, for a regular main tenant, do they have any non table data in their key space? I.e. if we replicate the key span `keys.MakeTenantSpan(3)`, will that only ever contain table data?

Reply:
> or a regular main tenant, do they have any non table data in their key space

No.

> replicate the key span keys.MakeTenantSpan(3), will that only ever contain table data

Correct.